core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -803,6 +816,7 @@ class ReplicaManager(val config: KafkaConfig,
   private def maybeShrinkIsr(): Unit = {
     trace("Evaluating ISR list of partitions to see which replicas can be removed from the ISR")
     allPartitions.values.foreach(partition => partition.maybeShrinkIsr(config.replicaLagTimeMaxMs))
+    maybePropagateIsrChanges()
```

Comment:
Instead of this, it seems it's simpler to just run maybePropagateIsrChanges() in a scheduler.